src/modules/kafka/kafka-container.ts

```diff
@@ -6,27 +6,48 @@ import { StartedTestContainer } from "../..";
 import { AbstractStartedContainer } from "../abstract-started-container";
 import { Host } from "../../docker/types";
 import { InspectResult } from "../../docker/functions/container/inspect-container";
-import { dockerClient } from "../../docker/docker-client";
 
 const KAFKA_PORT = 9093;
 const KAFKA_BROKER_PORT = 9092;
 const DEFAULT_ZOOKEEPER_PORT = 2181;
 
 export const KAFKA_IMAGE = "confluentinc/cp-kafka:5.5.4";
 
+export interface SaslSslListenerOptions {
+  sasl: SaslOptions;
+  port: number;
+  keystore: PKCS12CertificateStore;
+  truststore?: PKCS12CertificateStore;
+}
+
+export interface SaslOptions {
+  mechanism: "SCRAM-SHA-256" | "SCRAM-SHA-512";
+  user: User;
+}
+
+export interface User {
+  name: string;
+  password: string;
+}
+
+export interface PKCS12CertificateStore {
+  content: Parameters<GenericContainer["withCopyContentToContainer"]>[0];
```

Comment:
What does this mean? 🙂

Reply:
It means that this field accepts the same type as the first parameter of the `withCopyContentToContainer` method. Useful when we want to be secure for the future (in case the mentioned method changes), and if the method accepts a union of many types we don't want to copy all over the place :)